### ISSUE TITLE
Upgrade to Nuxt 3.15.0 led to 500 (Internal Server Error)

### ISSUE THREAD

**Issue by robmontesinos** [nuxt/nuxt#30384]
Immediately after upgrading to 3.15.0, and restarting dev server, I got the following error:

```
500
Unexpected token ';'

at createScript (node:vm:269:10)
at Object.runInThisContext (node:vm:317:10)
at ViteNodeRunner.runModule (./node_modules/.pnpm/vite-node@2.1.8_@types+node@22.10.2_terser@5.37.0/node_modules/vite-node/dist/client.mjs:398:19)
at ViteNodeRunner.directRequest (./node_modules/.pnpm/vite-node@2.1.8_@types+node@22.10.2_terser@5.37.0/node_modules/vite-node/dist/client.mjs:381:16)
at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
at async ViteNodeRunner.cachedRequest (./node_modules/.pnpm/vite-node@2.1.8_@types+node@22.10.2_terser@5.37.0/node_modules/vite-node/dist/client.mjs:206:14)
at async ViteNodeRunner.dependencyRequest (./node_modules/.pnpm/vite-node@2.1.8_@types+node@22.10.2_terser@5.37.0/node_modules/vite-node/dist/client.mjs:259:12)
at async ./node_modules/.pnpm/@headlessui+vue@1.7.23_vue@3.5.13_typescript@5.7.2_/node_modules/@headlessui/vue/dist/headlessui.esm.js:11:32
at async ViteNodeRunner.runModule (./node_modules/.pnpm/vite-node@2.1.8_@types+node@22.10.2_terser@5.37.0/node_modules/vite-node/dist/client.mjs:399:5)
``` 

IMHO a reproduction is not practical/possible for this error. 

The terminal error before the Server Error 500:

```
ℹ Using default Tailwind CSS file                                         nuxt:tailwindcss 2:56:27 PM
  ➜ DevTools: press Shift + Option + D in the browser (v1.7.0)                              2:56:30 PM

Hello from Tailwind.config                                                                  2:56:31 PM
ℹ Running with compatibility version 4                                                     2:56:32 PM
ℹ Tailwind Viewer: http://localhost:3000/_tailwind/                       nuxt:tailwindcss 2:56:32 PM
✔ Vite client built in 29ms                                                                2:56:34 PM
✔ Vite server built in 933ms                                                               2:56:35 PM
Hello from Tailwind.config                                                                  2:56:37 PM
✔ Nuxt Nitro server built in 2951 ms                                                 nitro 2:56:42 PM
Hello from Tailwind.config                                                                  2:56:42 PM
ℹ Vite client warmed up in 272ms                                                           2:56:42 PM

 ERROR  (node:97761) [DEP0040] DeprecationWarning: The punycode module is deprecated. Please use a userland alternative instead.
(Use node --trace-deprecation ... to show where the warning was created)

ℹ Vite server warmed up in 5639ms

```

I noticed the expected 'Vite client warmed up in x ms' did not appear.

Any suggestions? Should I revert back to previous version? Thank you in advance and Happy Holidays.

**Comment by danielroe** [comment]
this seems to be a vite issue (reported https://github.com/nuxt/ui/issues/2971)

you can temporarily resolve by pinning vite to v5

**Comment by robmontesinos** [comment]
Thank you @danielroe for the quick response. I tried pinning vite to 5 which didn't really work until I added to package.json
```
"pnpm": {
    "overrides": {
      "vite@6.0.6": "vite@5.4.11"
    }
```

Then it complained that Nuxt 3.15 cannot run with vite@5.4.11. So I followed @TheAlexLichter to downgrade Nuxt 3.14 and now the dev server is running again. I thank both of you for such a quick and effective response and wish you and your families and your team Merry Holidays and Happy New Year - salud!!
